### PR TITLE
CYS: Add homepage template AI completion and revamped header footer

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Part of https://github.com/woocommerce/woocommerce/issues/40162.

This PR:

- Merged header and footer into homepage template
- Added homepage template AI completion
- Added metadata into homepage template to assist with AI completion

### How to test the changes in this Pull Request:

This feature is part of the larger customize your store development and behind a feature flag, hence QA review can be deferred until the call for testing before release.

Make sure you are testing this with a Jetpack connected instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

### Test large business template

1. Create a new WooCommerce installation with this version.
2. Make sure to enable `customize-store` feature flag
3. Go to `WooCommerce -> Home` and click `Customize Store` task
4. Click `Design with AI` button
5. In the store description, use a description that signifies being a large business, for example: `My store sells modern furniture, we have factories in over 20 countries.`
6. Open browser's developer console, networks tab
7. Filter for `text-completion`
8. Proceed all steps until the end
9. Look for a query response that has `homepage_template` in it
10. Observe that the selected template is either `template1`, `template2`, or `template3`

### Test small business template

1. Go back to store description, use a description that signifies a small medium business, for example: `My store sells locally grown, homemade pies to all counties in a small state.`
8. Proceed all steps until the end
9. Look for a query response that has `homepage_template` in it
10. Observe that the selected template is either `template4`, `template5`, or `template6`
12. Should be redirected to assembler hub screen
13. Examine that the theme style and template are set based on AI response.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-homepage.tsx [hunk 1]
```diff
@@ -31,10 +31,7 @@ export const SidebarNavigationScreenHomepage = () => {
 	const [ blocks, , onChange ] = useEditorBlocks();
 	const onClickPattern = useCallback(
 		( _pattern, selectedBlocks ) => {
-			onChange(
-				[ blocks[ 0 ], ...selectedBlocks, blocks[ blocks.length - 1 ] ],
-				{ selection: {} }
-			);
+			onChange( selectedBlocks, { selection: {} } );
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-homepage.tsx]: Hey @ilyasfoo My understanding is on this screen, we show the homepage templates without header/footer.

@verofasulo Could you please confirm that when users select a homepage on this screen will replace the header/footer together? Thank you! 🙏 

Page list (3)

Comment (chihsuan) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-homepage.tsx]: Attach the current changes's recording.

https://github.com/woocommerce/woocommerce/assets/4344253/8ca2fa5b-581a-4325-85d0-a4df26c22219

Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-homepage.tsx]: Ah, I can't believe I missed this fact 🤦 

I'll fix it next week

Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-homepage.tsx]: @chihsuan I've fixed it in https://github.com/woocommerce/woocommerce/pull/40363/commits/b1ce8b108f41f7335c6dbc4c0987b5be276378f3 added tests in https://github.com/woocommerce/woocommerce/pull/40363/commits/68ce0211141121819882cbfb53613047fd7efd49. Please take another look! 🙏

**Review by chihsuan — APPROVED** [review]
Great work on this! LGTM and tested well. 🚀

**Review by rjchow — APPROVED** [review]
LGTM! Seems to work correctly with a few different descriptions I tried

**Pull request merged**